### ISSUE TITLE
Test breakages with OpenSSL 1.1.0

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8898]
|[<img alt="freeekanayaka's avatar" src="https://avatars.githubusercontent.com/u/598445?s=50" width="50" height="50">](https://github.com/freeekanayaka)| @freeekanayaka reported|
|-|-|
|Trac ID|trac#8898|
|Type|release blocker: regression|
|Created|2016-11-10 09:18:17Z|

Unit tests break when run wih OpenSSL 1.1.0b2 (the current version in Debian/sid).

The offending changes with respect to OpenSSL 1.0 seem to be:

- SSLv2 support got dropped
- keys are required to be at least 4096 bytes long

I'm attaching a stop-the-gap patch, that will most probably need some curation.

See at the bottom of this log:

https://api.travis-ci.org/jobs/174481746/log.txt?deansi=true

for details about the test failures.

Attachments:

* [0007-openssl-1.1.0.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/273/2734410677a5bab3c9d09774226ccd8f0e6b4e0a/f5dddbc1ddca6218eb8b22e9a1fcecd026de75cf.patch) (10269 bytes) - added by freeekanayaka on 2016-11-10 09:19:26Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8898 8898
type__release_blocker__regression release blocker: regression
reporter__freeekanayaka freeekanayaka
priority__normal normal
milestone__None None
branch__None None
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1478769497823185 1478769497823185
changetime__1490020174094615 1490020174094615
version__None None
owner__markrwilliams markrwilliams

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @markrwilliams|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_3"></a>|@markrwilliams commented|
|-|-|

https://github.com/twisted/twisted/pull/743

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams commented|
|-|-|

Reviews:

https://github.com/twisted/twisted/pull/743#pullrequestreview-26441405
https://github.com/twisted/twisted/pull/743#pullrequestreview-26445480 

Approval:
https://github.com/twisted/twisted/pull/743#pullrequestreview-27759006

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_6"></a>|@markrwilliams set status to `closed`|
|-|-|

In changeset 79ae40097eb280ce2aa2c81521d4c9c758d12921
```
#!CommitTicketReference repository="" revision="79ae40097eb280ce2aa2c81521d4c9c758d12921"
Merge pull request #743 from twisted/8898-openssl-110

Fix tests broken by OpenSSL 1.1.0

Author: Free Ekanayaka, markrwilliams
Reviewers: Lukasa, alex
Fixes: ticket:8898
```
